src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts

```diff
@@ -0,0 +1,391 @@
+                enrichedComplex.startSubComponentIndex = this.getMinValue(enrichedComplex.startSubComponentIndex, k);
+                enrichedComplex.endSubComponentIndex = this.getMaxValue(enrichedComplex.endSubComponentIndex, k);
+              }
+            }
+          }
+        }
+      }
+    }
+
+    // We finally check the position of the subcomponents of subcomplexes part of the complex on the main table.
+    // If any is before or after where the lines start and end, then we need to update the start and end indexes,
+    // as the line must start or end there
+    for (let i = 0; i < this._enrichedInteractors.length; i++) {
+      if (!this._enrichedInteractors[i].hidden) {
+        if (subComponentsToCheck.includes(this._enrichedInteractors[i].interactor.identifier)) {
+          // The interactor is part of a subcomplex that is part of the complex, we update the start and end indices for the interactors
+          // line as it may start in this interactor
+          enrichedComplex.startInteractorIndex = this.getMinValue(enrichedComplex.startInteractorIndex, i);
+          enrichedComplex.endInteractorIndex = this.getMaxValue(enrichedComplex.endInteractorIndex, i);
+        }
+      }
+    }
+    return enrichedComplex;
+  }
+
+  private interactorOrganism() {
+    for (const complex of this.complexSearch.elements) {
+      const organismName = complex.organismName;
+      for (const complexInteractor of complex.interactors) {
+        // tslint:disable-next-line:max-line-length
+        const match = this._enrichedInteractors.find(enrichedInteractor => enrichedInteractor.interactor.identifier === complexInteractor.identifier);
+        match.organismName = organismName;
+      }
+    }
+  }
+
+  public classifyInteractorsByOrganism() {
+    this._enrichedInteractors.sort((a, b) => b.organismName.localeCompare(a.organismName));
+    // this.calculateAllStartAndEndIndexes();
+    this.rangeOfInteractorOrganismV2();
+  }
+
+  public classifyInteractorsByType() {
+    this._enrichedInteractors.sort((a, b) => b.interactor.interactorType.localeCompare(a.interactor.interactorType));
+    // this.calculateAllStartAndEndIndexes();
+    this.rangeOfInteractorTypeV2();
+  }
+
+  public classifyInteractorsByOccurrence() {
+    for (const oneInteractor of this._enrichedInteractors) {
+      for (const complex of this.complexSearch.elements) {
+        for (const complexesInteractors of complex.interactors) {
+          if (oneInteractor.interactor.identifier === complexesInteractors.identifier) {
+            // tslint:disable-next-line:radix
+            if (isNaN(parseInt(this.stoichiometryOfInteractors(complex, oneInteractor.interactor.identifier)))) {
+              oneInteractor.timesAppearing = oneInteractor.timesAppearing;
+            } else {
+              oneInteractor.timesAppearing += parseInt(this.stoichiometryOfInteractors(complex, oneInteractor.interactor.identifier), 10);
+            }
```

Comment:
```
if (!isNaN(parseInt(this.stoichiometryOfInteractors(complex, oneInteractor.interactor.identifier)))) {
              oneInteractor.timesAppearing += parseInt(this.stoichiometryOfInteractors(complex, oneInteractor.interactor.identifier), 10);
            }
```

Reply:
Actually, instead of calling `this.stoichiometryOfInteractors` twice, we could do:
```
const stoichiometryValue = parseInt(this.stoichiometryOfInteractors(complex, oneInteractor.interactor.identifier), 10);
if (!isNaN(stoichiometryValue)) {
    oneInteractor.timesAppearing += stoichiometryValue;
}
```